server/tso/allocator_manager.go

```diff
@@ -451,6 +476,27 @@ func (am *AllocatorManager) ClusterDCLocationChecker() {
 			continue
 		}
 		am.mu.clusterDCLocations[dcLocation] = append(am.mu.clusterDCLocations[dcLocation], serverID)
+		// Only the PD leader can check and update the sortedDCLocations
+		if am.member.IsStillLeader() {
+			// Add the new dc-location found from the etcd into the sortedDCLocations
+			if slice.NoneOf(am.mu.sortedDCLocations, func(i int) bool {
+				return am.mu.sortedDCLocations[i] == dcLocation
+			}) {
+				am.mu.sortedDCLocations = append(am.mu.sortedDCLocations, dcLocation)
+			}
+		}
```

Comment:
After `transfer pd leader`, can we maintain the order like before? I think this will be quite useful.